Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -0,0 +1,357 @@
+/-
+Copyright (c) 2023 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+import Mathlib.Topology.Bases
+import Mathlib.Order.Filter.CountableInter
+import Mathlib.Topology.Compactness.Compact
+
+/-!
+# Lindelöf sets and Lindelöf spaces
+
+## Main definitions
+
+We define the following properties for sets in a topological space:
```

Comment:
The module docstring is a good place to discuss equivalent definitions of the notions defined in this file, explain why these notions are useful etc. In this case, you should

- Say that Lindelöf sets/spaces can be defined via filters or via open subcovers. Currently, it looks like you're using 2 different definitions for `IsLindelof` and `LindelofSpace`.
- The API is based on the API of `IsCompact` (e.g., in the "Implementation Notes" section).
- A second-countable topological space is strongly Lindelöf.
- A countable set is Lindelöf; more generally, a countable union of Lindelöf sets is Lindelöf.
- A σ-compact space is Lindelöf and a locally compact Lindelöf space is σ-compact.
- Give or reference an example of a non-Lindelöf set.

I know that many existing module level docstrings are very short but we're trying to improve the documentation, starting with new files. The goal is to make the module docstrings readable for mathematicians who never saw the concepts discussed in the file before.

Reply:
Thanks, I was not aware of that indeed. Most of these results are not yet in the PR, so I'll write in the module docstring what I have.